connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/executable/CancellationManager.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.runtime.inbound.executable;
+
+import io.camunda.connector.api.error.ConnectorRetryException;
+import io.camunda.connector.api.inbound.Health;
+import java.time.Duration;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handles exceptions thrown during the cancellation process of an {@link
+ * io.camunda.connector.api.inbound.InboundConnectorContext#cancel(Throwable) cancel} operation.
+ * This class determines the appropriate action to take based on the specific exception encountered.
+ */
+public class CancellationManager {
+
+  private static final Logger LOG = LoggerFactory.getLogger(CancellationManager.class);
+  private final Map<UUID, RegisteredExecutable> executables;
```

Comment:
I'm not a huge fan of having access to executables state from two different classes (`CancellationManager` and `InboundExecutableRegistryImpl`). This way it's less transparent where this state is changed.

One idea how we could tackle this:
- Instead of an external `CancellationManager`, we could introduce a proxy wrapper over the executable (e.g. `StatefulExecutable` or smth along those lines). I guess it can also implement the `InboundConnectorExecutable` interface.
- This proxy wrapper can manage the lifecycle of the executable inside of itself. It can store the real executable as a private field and create a cancellation callback; when the real executable calls the `cancel` method, a cancellation callback will trigger a method inside the proxy wrapper that will lead to the recreation of the executable.
- The proxy wrapper can also keep track of the state (thus we can probably get rid of `RegisteredExecutable`).
- The rest of the connector runtime doesn't know anything about retries and the `executables` Map in the `InboundExecutableRegistry` doesn't need to be modified anywhere outside of the registry class.

WDYT? I might have missed some important details, so happy to brainstorm together.